### PR TITLE
Contacts profile redirects to setting page, updates and deletion not …

### PR DESCRIPTION
**Pull request by svogt0511**
…possible

## Purpose

closes: #568, #537, datacite/datacite#1449

Needs: datacite/lupo#930 for the delete button to work.

## Approach

When logged in to Fabrica as a consortium admin, clicking a contact from the list in the contacts tab redirects to the settings page instead of showing contact details.  Unable to see individual contact details, update or delete contacts.  This happens because the access (abilities) for consortium_lead accounts was determined incorrectly

Note that delete contact button is limited.  If there are roles associated with a contact, they must be removed first.  If the contact is created in the consortium by the consortium lead, it can be deleted by the consortium lead.

Vercel preview: https://bracco-4h01pwfar-datacite.vercel.app/

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)

- [ ] New feature (non-breaking change which adds functionality)

- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Reviewer, please remember our [guidelines](https://datacite.atlassian.net/wiki/spaces/TEC/pages/1168375809/Pull+Request+Guidelines):

- Be humble in the language and feedback you give, ask don't tell.
- Consider using positive language as opposed to neutral when offering feedback. This is to avoid the negative bias that can occur with neutral language appearing negative.
- Offer suggestions on how to improve code e.g. simplification or expanding clarity.
- Ensure you give reasons for the changes you are proposing.

### CODE REVIEW

**Review by digitaldogsbody — APPROVED** [review]

**Comment by digitaldogsbody** [comment]
Are there tests in the suite for this functionality?

File: app/abilities/contact.js [hunk 1]
```diff
@@ -17,8 +17,10 @@ export default Ability.extend({
             return true;
           case 'consortium_admin':
             return (
-              this.get('currentUser.provider_id') ===
-              this.get('model.provider.consortium.id')
+              ((this.get('model.provider.memberType') === 'consortium_organization') &&
```
Comment (jrhoads) [inline app/abilities/contact.js:20]: This big conditional block is repeated 4 times.  Can we refactor this into a helper function so we can have the logic in one place if we need to understand/update it in the future.

**Review by digitaldogsbody — APPROVED** [review]

File: app/abilities/contact.js (hunk 1 above)
Comment (svogt0511) [reply app/abilities/contact.js:20]: @jrhoads I have moved that code to a utility 'isAbleTo'.

Comment (jrhoads) [reply app/abilities/contact.js]: Looks good 👍

**Review by jrhoads — APPROVED** [review]

**Pull request merged**